tests/EventListener/WriteListenerTest.php

```diff
@@ -108,7 +108,8 @@ public function testOnKernelViewWithControllerResultAndPersistWithImmutableResou
         $dataPersisterProphecy->supports($dummy, Argument::type('array'))->willReturn(true)->shouldBeCalled();
 
         $iriConverterProphecy = $this->prophesize(IriConverterInterface::class);
-        $iriConverterProphecy->getIriFromItem($dummy)->willReturn('/dummy/1')->shouldBeCalled();
+        $iriConverterProphecy->getIriFromItem($dummy)->shouldNotBeCalled();
+        $iriConverterProphecy->getIriFromItem($dummy2)->willReturn('/dummy/1')->shouldBeCalled();
```

Comment:
Do an assert on the `Request` attributes instead. :smile:

Reply:
I'm sorry, I don't follow, the above assertion checks whether or not the right call is done on the IriConverter. Since the same Iri is returned the current assertion on the request is still valid.

Reply:
Yes, what @teohhanhui means is that it'd be nice to check that `$request->attributes->set('_api_write_item_iri` has the expected IRI (https://github.com/api-platform/core/blob/3f1f23fcdabd695f2d679300f41f79c8e4d66ee8/src/EventListener/WriteListener.php#L81) because this was the root issue

Reply:
Yes, I agree, but that's already asserted :) That didn't actually change because Prophecy is returning the same IRI but only for another Dummy (`dummy2`) resource.

Reply:
`$dummy` and `$dummy2` should have different IRI for an effective test.

Reply:
@Tjeerd I'll take care of the test. :smile: